code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -251,8 +251,10 @@ mob/living/simple_animal/hostile/hitby(atom/movable/AM as mob|obj,var/speed = TH
 		return FALSE
 
 	var/target_hit = FALSE
-
-	for(var/mob/M in check_trajectory(target_mob, src, pass_flags=PASSTABLE))
+	var/flags = PASSTABLE|PASSGLASS|PASSGRILLE
+	if(istype(projectiletype, /obj/item/projectile/beam))

```

Comment:
I would have said, use the projectile's own `pass_flags` parameter to get their flags. But fuck, this isn't available as an instance right now, and that is hurtsome. About as hurtsome as the hardcoding of types like this.